src/main/java/io/rainfall/statistics/monitor/MemStatisticsCollector.java

```diff
@@ -0,0 +1,115 @@
+package io.rainfall.statistics.monitor;
+
+import io.rainfall.reporting.HtmlReporter;
+import io.rainfall.statistics.collector.StatisticsCollector;
+import io.rainfall.statistics.exporter.Exporter;
+import io.rainfall.statistics.exporter.HtmlExporter;
+import io.rainfall.statistics.exporter.TextExporter;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.Writer;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * Created by KECL on 8/18/2016.
+ */
+public class MemStatisticsCollector  implements StatisticsCollector {
+
+  private static final MemoryMXBean MEM_BEAN = ManagementFactory.getMemoryMXBean();
+  private static final String MEM_STATS = "Memory Utilization";
+
+  private Writer memOutput;
+
+  public static class MemStats {
+    public enum Header {
+      MEMORY_UTILIZATION
+    }
+
+    private final long startTimestamp;
+    private final long memUtilization;
+
+    public MemStats(long startTimestamp, long memUtilization) {
+      this.startTimestamp = startTimestamp;
+      this.memUtilization = memUtilization;
+    }
+  }
+
+  private final Queue<MemStatisticsCollector.MemStats> memStatsQueue = new ConcurrentLinkedQueue<MemStatisticsCollector.MemStats>();
+
+  @Override
+  public void initialize() {}
+
+  @Override
+  public void terminate() {}
+
+  @Override
+  public Exporter peek() {
+    memStatsQueue.add(new MemStatisticsCollector.MemStats(System.currentTimeMillis(), MEM_BEAN.getHeapMemoryUsage().getUsed()));
```

Comment:
you don't need a Queue here, you can pass as a constructor of the Exporter the Heap usage value (and probably a timestamp too)